This fixes issues with build-dependencies that are varying across
python version:

- python-argparse, which is no necessary for python >= 2.7
- python-mock, which is not necessary for python >= 3.3
  Additionally mock is a sub-package of unittest since it is
  part of CPython, which means import have to be modified.
    
Those problems did not appeared before because upstream distributes
the package only as a source package.
--- a/setup.py
+++ b/setup.py
@@ -14,6 +14,20 @@
 
 __version__ = '1.3'
 
+
+def install_requirements():
+    requirements = [
+        'nose',
+        'Pygments==2.0.1',
+        ]
+    if (2, 7, 0) > sys.version_info:
+        requirements.append('argparse')
+
+    if (3, 3, 0) > sys.version_info:
+        requirements.append('mock')
+
+    return requirements
+
 # The docs import setup.py for the version so only call setup when not behaving
 # as a module.
 if __name__ == '__main__':
@@ -22,13 +36,7 @@
 
     long_description = __doc__ + '\n\n' + releases
 
-    install_requires = [
-        'argparse',  # For Python 2.6 support
-        'mock',
-        'nose',
-        'Pygments==2.0.1',
-    ]
-
+    install_requires = install_requirements()
     # Add some developer tools.
     if 'develop' in sys.argv:
         install_requires.extend([
--- a/tap/tests/test_adapter.py
+++ b/tap/tests/test_adapter.py
@@ -2,7 +2,10 @@
 
 import sys
 
-import mock
+try:
+    from unittest import mock
+except ImportError:
+    import mock
 
 from tap.adapter import Adapter
 from tap.tests import TestCase
